client/app/core/navigation.service.js

```diff
@@ -81,7 +91,8 @@ export function NavigationFactory (RBAC, Polling, POLLING_INTERVAL, CollectionsA
     menuItems.forEach((item) => {
       if (angular.isDefined(item.badges)) {
         getBadgeCount(item.badgeQuery.field, item.badgeQuery.filter).then((count) => {
-          item.badges[0].count = count
+          service.addCount(item.state, count)
+          item.badges[0].count = service.navCount[item.state]
```

Comment:
will this re-trigger when a value change? e.g. is there a direct relationship between a notify event on the redux state to this function?

Reply:
@ohadlevy This is currently not reactive in the way you described.  Currently the navigation functionality for getting counts is a polling function.  The polling system calls this function  and it updates the different counts in a loop.

Reply:
You'll probably want to converge to a system where the state of a unit (component, controller, service, factory, etc.) is a transformation of the whole app state, managed through the Redux store.

The app state (in Redux store) should be the single source of "truth" - whenever the state changes, all relevant units should be notified and updated according to the new data. You can consider using libraries like `reselect` to optimize this process, e.g. avoid component update if the particular bits of data relevant for that component didn't change (but other bits of data did change).

In a pure Redux architecture, a unit doesn't have its own state - all the data is managed through the Redux store. Change of the app state happens by firing actions (actions flow up) which cause state modification which then causes all subscribers to be notified (data flow down).

But I understand that adopting pure Redux arch can be challenging so you can improve step by step. The main strength of Redux is that you know exactly what the current app data is (because it's in one place, not dispersed across X units) and how exactly does it change (because you have your reducers dedicated for this purpose).

Reply:
@chalettu what would it take to make sure this value is stored only in redux?

Reply:
@ohadlevy , I just made changes so that the count for each menu item is retrieved from the redux store when the menu is defined.  The other code that is already in this PR handles what happens when polling occurs.  All code for nav counts now only uses redux for setting and retrieving counts.

Reply:
in theory, I would the action to trigger fetching the API, meaning that the logic is initiated from the action and not the other way around